backend/company-agent/src/main/java/org/hyperledger/oa/repository/DBOperations.java

```diff
@@ -39,18 +45,36 @@
     @Inject
     private UserRepository userRepo;
 
+    @Inject
+    Optional<SchemaService> schemaService;
+
     @Value("${oagent.bootstrap.username}")
     private String username;
 
     @Value("${oagent.bootstrap.password}")
     private String password;
 
+    @Property(name = "oagent.schemes")
+    List<Map<String, String>> schemes;
+
     @Async
     @EventListener
     public void onServiceStartedEvent(StartupEvent startupEvent) {
         log.info("Running startup database operations.", startupEvent);
 
-        // create default user
+        createDefaultUser();
+        createDefaultSchemes();
+
+        log.debug("Done running database operations.");
+    }
+    
+    private void createDefaultSchemes() {
+    	log.debug("Purging and re-setting default schemes.");
+    	
+        schemaService.get().setWriteOnlySchemes(schemes);
```

Comment:
Broken in web only mode. This is on conditional Bean you always have to check for existence first.